### ISSUE TITLE
Multisite connection fails with 404 error

### ISSUE THREAD

**Issue by iandunn** [Automattic/jetpack#13109]
Starting within the past few weeks, I'm getting a `Site connection failed!` error when trying to connect a site to WPCOM from the Multisite sites list (`wp-admin/network/admin.php?page=jetpack`)

`jetpack_sites_list()` calls `do_subsiteregister()`, which [makes a remote request](https://github.com/Automattic/jetpack/blob/3ed3de8/class.jetpack-network.php#L506) to https://jetpack.wordpress.com/jetpack.subsiteregister/1/ which then responds with a `404`.

if I `var_dump( $response )`, I get:

```
array (size=6)
  'headers' => 
    object(Requests_Utility_CaseInsensitiveDictionary)[2390]
      protected 'data' => 
        array (size=10)
          'server' => string 'nginx' (length=5)
          'date' => string 'Mon, 22 Jul 2019 19:19:58 GMT' (length=29)
          'content-type' => string 'application/json' (length=16)
          'vary' => string 'Accept-Encoding' (length=15)
          'x-hacker' => string 'If you're reading this, you should visit automattic.com/jobs and apply to join the fun, mention this header.' (length=108)
          'expires' => string 'Wed, 11 Jan 1984 05:00:00 GMT' (length=29)
          'cache-control' => string 'no-cache, must-revalidate, max-age=60' (length=37)
          'content-encoding' => string 'gzip' (length=4)
          'x-ac' => string '4.ord _dca' (length=10)
          'strict-transport-security' => string 'max-age=15552000' (length=16)
  'body' => string '' (length=0)
  'response' => 
    array (size=2)
      'code' => int 404
      'message' => string 'Not Found' (length=9)
  'cookies' => 
    array (size=0)
      empty
  'filename' => null
  'http_response' => 
    object(WP_HTTP_Requests_Response)[2386]
      protected 'response' => 
        object(Requests_Response)[2387]
          public 'body' => string '' (length=0)
          public 'raw' => string 'HTTP/1.1 404 Not Found

Server: nginx

Date: Mon, 22 Jul 2019 19:19:58 GMT

Content-Type: application/json

Transfer-Encoding: chunked

Connection: close

Vary: Accept-Encoding

X-hacker: If you're reading this, you should visit automattic.com/jobs and apply to join the fun, mention this header.

Expires: Wed, 11 Jan 1984 05:00:00 GMT

Cache-Control: no-cache, must-revalidate, max-age=60

Content-Encoding: gzip

X-ac: 4.ord _dca 

Strict-Transport-Security: max-age=15552000



' (length=482)
          public 'headers' => 
            object(Requests_Response_Headers)[2388]
              protected 'data' => 
                array (size=10)
                  'server' => 
                    array (size=1)
                      0 => string 'nginx' (length=5)
                  'date' => 
                    array (size=1)
                      0 => string 'Mon, 22 Jul 2019 19:19:58 GMT' (length=29)
                  'content-type' => 
                    array (size=1)
                      0 => string 'application/json' (length=16)
                  'vary' => 
                    array (size=1)
                      0 => string 'Accept-Encoding' (length=15)
                  'x-hacker' => 
                    array (size=1)
                      0 => string 'If you're reading this, you should visit automattic.com/jobs and apply to join the fun, mention this header.' (length=108)
                  'expires' => 
                    array (size=1)
                      0 => string 'Wed, 11 Jan 1984 05:00:00 GMT' (length=29)
                  'cache-control' => 
                    array (size=1)
                      0 => string 'no-cache, must-revalidate, max-age=60' (length=37)
                  'content-encoding' => 
                    array (size=1)
                      0 => string 'gzip' (length=4)
                  'x-ac' => 
                    array (size=1)
                      0 => string '4.ord _dca' (length=10)
                  'strict-transport-security' => 
                    array (size=1)
                      0 => string 'max-age=15552000' (length=16)
          public 'status_code' => int 404
          public 'protocol_version' => float 1.1
          public 'success' => boolean false
          public 'redirects' => int 0
          public 'url' => string 'https://jetpack.wordpress.com/jetpack.subsiteregister/1/' (length=56)
          public 'history' => 
            array (size=0)
              empty
          public 'cookies' => 
            object(Requests_Cookie_Jar)[2385]
              protected 'cookies' => 
                array (size=0)
                  empty
      protected 'filename' => null
      public 'data' => null
      public 'headers' => null
      public 'status' => null
```

The Debug screen says everything is fine.

At a quick glance, possibly related #10971, #12325, 3ed3de86a58730377bbc29f8b7df7aa7729f28ce

**Comment by iandunn** [comment]
Actually, now the Debug screen says `Jetpack.com detected an error. Visit the Jetpack.com debugging page for more information or contact support.`

...but it doesn't link to that page, and I don't see anything. Where is that?

**Comment by iandunn** [comment]
If I try connecting from the individual site rather than the network settings, I get this error:

> Error Details: The Jetpack server could not communicate with your site's XML-RPC URL. If you have the W3 Total Cache plugin installed, deactivate W3 Total Cache, try to Connect to WordPress.com again, reactivate W3 Total Cache, then clear W3 Total Cache's cache.

W3 Total Cache is not installed, and  XML-RPC seems fine:

```sh
curl -d "<?xml version=\"1.0\"?> <methodCall> <methodName>system.listMethods</methodName> </methodCall>" https://2020.atlanta.wordcamp.org/xmlrpc.php
```

**Comment by iandunn** [comment]
It turns out that 7.5.0 broke connecting via XMLRPC, but I'm assuming there's another issue for that.

When XMLRPC isn't available, Jetpack tries to connect over the REST API, but it turns out that my site had accidentally disabled the Jetpack endpoints, which caused that to fail.

**Comment by jeherve** [comment]
> It turns out that 7.5.0 broke connecting via XMLRPC, but I'm assuming there's another issue for that.

Yes, this will be fixed in #13139
